File: packages/android/src/main/java/com/kozmos/components/Search/Search.figma.kt
package com.kozmos.components.search

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=80-391")
class KozmosSearchConnect {
    @FigmaProperty(FigmaType.Text, "Placeholder Text")
    val placeholder: String = "Search"

    @Composable
    fun ComponentExample() {
        KozmosSearch(
            value = "",
            onValueChange = {},
            placeholder = placeholder
        )
    }
}

File: packages/ios/Sources/Components/Search/Search.figma.swift
import SwiftUI
import Figma

struct KozmosSearchConnect: FigmaConnect {
    let component = KozmosSearch.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=80-391"

    @FigmaString("Placeholder Text")
    var placeholder: String = "Search"

    var body: some View {
        KozmosSearch(
            text: .constant(""),
            placeholder: self.placeholder
        )
    }
}
